dashboard/api.py

```diff
@@ -293,19 +294,53 @@ def get_status_for_courserun(course_run, mmtrack):
     if mmtrack.is_enrolled_mmtrack(course_run.edx_course_key):
         if course_run.is_current:
             status = CourseRunStatus.CURRENTLY_ENROLLED
-        elif course_run.is_past:
-            status = CourseRunStatus.CHECK_IF_PASSED
         elif course_run.is_future:
             status = CourseRunStatus.WILL_ATTEND
-        # If a course run has no start or end date, is_past, is_current, and is_future all return False
-    else:
-        if course_run.is_past:
-            status = CourseRunStatus.NOT_PASSED
+        elif course_run.has_frozen_grades:
+            status = CourseRunStatus.CHECK_IF_PASSED
+        # this last check needs to be done as last one
+        elif course_run.is_past:
+            status = CourseRunStatus.CURRENTLY_ENROLLED
         else:
+            raise ImproperlyConfigured(
+                'The course {0} results not being current, past or future at the same time'.format(
```

Comment:
will go with `The course {0} results are not either current, past, or future at the same time`
is it ok?